### PR TITLE
Fix exception for invalid api key

### PR DESCRIPTION
**Pull request by klaasnicolaas**
This will add status code 403 to RequestError exception, because error messages related to the api key do not fall under error 400 but 403.

### CODE REVIEW

File: forecast_solar/__init__.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ async def _request(
         if response.status < 500:
             self.ratelimit = Ratelimit.from_response(response)
 
-        if response.status == 400:
+        if response.status in [400, 403]:
```
Comment (MartinHjelmare) [inline forecast_solar/__init__.py:112]: We don't seem to handle any of these library specific exceptions in the integration. We may want to raise a specific exception if authentication fails, so the library user can ask the end user to re-authenticate.

Comment (klaasnicolaas) [reply forecast_solar/__init__.py:112]: Something like this? 

```py
class ForecastSolarAuthenticationError(ForecastSolarError):
    """Forecast.Solar API authentication exception."""

    def __init__(self, data: dict) -> None:
        """Init a solar auth error.

        https://doc.forecast.solar/doku.php?id=api#invalid_request
        """
        super().__init__(f'{data["text"]} (error {data["code"]})')
        self.code = data["code"]
```

Comment (MartinHjelmare) [reply forecast_solar/__init__.py:112]: Yes. We normally start a reauth flow by raising `ConfigEntryAuthFailed` in the integration. See existing integrations that do this.

Comment (klaasnicolaas) [reply forecast_solar/__init__.py]: image

I think only if you actually went over the rate limit?

Comment (MartinHjelmare) [reply forecast_solar/__init__.py]: Ok, so maybe move that code under that specific exception and remove the general 500 case.

Comment (klaasnicolaas) [reply forecast_solar/__init__.py]: Maybe we should save this for another pull request and see what we want to do with the ratelimit in the end. Currently it is only used in the example.py at a status of 200 to see what your limit is. At status 429, which is also about the rate limit, you will see other data.

Comment (MartinHjelmare) [reply forecast_solar/__init__.py]: Ok.

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**